drivers/watchdog/wdt_iwdg_stm32.c

```diff
@@ -174,7 +188,20 @@ static int iwdg_stm32_install_timeout(const struct device *dev,
 	/* Store the calculated values to write in the iwdg registers */
 	data->prescaler = prescaler;
 	data->reload = reload;
+#ifdef CONFIG_IWDG_STM32_HAS_EARLY_WAKEUP
+	LL_IWDG_EnableWriteAccess(cfg->instance);
+	uint32_t ewi_time = EWI_TIME;
 
+	if (ewi_time >= reload) {
+		if (reload <= 2U) {
+			/* Reload value is too small and EWI time cannot be set */
+			return -EINVAL;
+		}
+		/* EWI time should be less than reload time but should be at least 2 */
+		ewi_time = (reload > 2U) ? (reload - 1U) : 2U;
+	}
```

Comment:
1. Create a Kconfig option for the early wake-up threshold instead of hardcoding it
2. Simplify the logic + rewording
```suggestion
        if (reload == 1U) { //i.e., < 2U
            /* Early wake-up is not possible if counter is
             * reloaded with value 1 as the system will be
             * reset immediately when the counter decrements.
             * (Note: reload=1 is NOT RECOMMENDED per RefMan)
             */
             return -EINVAL;       
        }

        /*
         * If Kconfig is higher than reload, set early wake-up
         * as high as possible (= reload - 1); otherwise, set
         * to the requested value.
         */
	uint32_t ewi_time = MIN(CONFIG_EWI_TIME, reload);

	if (ewi_time >= reload) {
		if (reload <= 2U) {
			/* Reload value is too small and EWI time cannot be set */
			return -EINVAL;
		}
		/* EWI time should be less than reload time but should be at least 2 */
		ewi_time = (reload > 2U) ? (reload - 1U) : 2U;
	}
```

Note: as far as I understand, it is valid to have `RLR = EWIT` (since early wake-up occurs when counter is *strictly less than* `EWIT`)?

Note 2: Kconfig should be `range 2 4095`